Userland/Libraries/LibCore/LocateDb.cpp

```diff
@@ -0,0 +1,43 @@
+#include <LibCore/LocateDb.h>
+
+namespace Core {
+
+LocateDb::LocateDb(String path, LocateDbMode mode)
+{
+    file_handle = fopen(path.characters(), (mode == LocateDbMode::write ? "w" : "r"));
+    byte_offset = 0;
+}
+
+LocateDb::~LocateDb()
+{
+    fclose(file_handle);
+}
+
+void LocateDb::write_header()
+{
+    // Format: magic string + version.
+    String header = String::format("%s|%d\n", "locatedb", 1);
+    fwrite(header.characters(), sizeof(char), header.length(), file_handle);
```

Comment:
You can replace these two lines with this:
```c++
outln(file_handle, "{}|{}", "locatedb", 1);
```

Also, please prefer the newer AK/Format functions over the printf-style ones (this isn't a hard requirement, so if at some point you feel like using printf-style formatting would be more clear, feel free to use it)